source/core_pkcs11.c

```diff
@@ -295,19 +275,30 @@ CK_RV xInitializePkcs11Session( CK_SESSION_HANDLE * pxSession )
     }
 
     /* Open a PKCS #11 session. */
-    if( ( xResult == CKR_OK ) && ( pxSlotId != NULL ) && ( xSlotCount >= 1UL ) )
+    if( ( xResult == CKR_OK ) && ( xSlotCount >= 1UL ) )
```

Comment:
( pxSlotId != NULL ) is a redundant chech since the value xResult ensure this won't be NULL.